editor.planx.uk/src/@planx/components/Calculate/Editor.tsx

```diff
@@ -36,12 +33,49 @@ export default function Component(props: Props) {
         props.handleSubmit({ type: TYPES.Calculate, data: newValues });
       }
     },
-    validate: () => {
-      // can parse formula
-      getVariables(formik.values.formula);
+    validate: (values) => {
+      const errors: FormikErrors<Calculate> = {};
+      try {
+        // can parse formula
+        getVariables(formik.values.formula);
+
+        // Validate formula
+        const result = evaluate(
+          values.formula,
+          values.samples,
+          values.defaults,
+        );
+
+        if (Number.isNaN(Number(result))) {
```

Comment:
As mentioned in the description, I have used this to catch issues with the `NumberInput` component which existed but wasn't picked up because we were never blocking the update of the comp